compile failed with msys2

../../third_party/loguru\loguru.hpp:1636:16: error: use of undeclared identifier 'vasprintf'; did you mean 'vsprintf'?
                int result = vasprintf(&buff, format, vlist);
                             ^
D:\msys64\mingw64\x86_64-w64-mingw32\include\stdio.h:372:5: note: 'vsprintf' declared here
int vsprintf (char *__stream, const char *__format, __builtin_va_list __local_argv)
    ^
In file included from ../../src/third_party_impl.cc:5:
../../third_party/loguru\loguru.hpp:1636:26: error: cannot initialize a parameter of type 'char *' with an rvalue of type 'char **'
                int result = vasprintf(&buff, format, vlist);
                                       ^~~~~
D:\msys64\mingw64\x86_64-w64-mingw32\include\stdio.h:372:21: note: passing argument to parameter '__stream' here
int vsprintf (char *__stream, const char *__format, __builtin_va_list __local_argv)
                    ^
In file included from ../../src/third_party_impl.cc:5:
../../third_party/loguru\loguru.hpp:1878:3: error: use of undeclared identifier 'localtime_r'; did you mean 'localtime_s'?
                localtime_r(&sec_since_epoch, &time_info);
                ^
D:\msys64\mingw64\x86_64-w64-mingw32\include\time.h:261:31: note: 'localtime_s' declared here
__forceinline errno_t __cdecl localtime_s(struct tm *_Tm,const time_t *_Time) { return _localtime64_s(_Tm,_Time); }
                              ^
In file included from ../../src/third_party_impl.cc:5:
../../third_party/loguru\loguru.hpp:1878:15: error: cannot initialize a parameter of type 'struct tm *' with an rvalue of type 'time_t *' (aka 'long long *')
                localtime_r(&sec_since_epoch, &time_info);
                            ^~~~~~~~~~~~~~~~
D:\msys64\mingw64\x86_64-w64-mingw32\include\time.h:261:54: note: passing argument to parameter '_Tm' here
__forceinline errno_t __cdecl localtime_s(struct tm *_Tm,const time_t *_Time) { return _localtime64_s(_Tm,_Time); }
                                                     ^
In file included from ../../src/third_party_impl.cc:5:
../../third_party/loguru\loguru.hpp:1946:8: error: no matching function for call to 'mkdir'
                        if (mkdir(file_path, 0755) == -1) {
                            ^~~~~
D:\msys64\mingw64\x86_64-w64-mingw32\include\io.h:280:15: note: candidate function not viable: requires 1 argument, but 2 were provided
  int __cdecl mkdir (const char *) __MINGW_ATTRIB_DEPRECATED_MSVC2005;
              ^
In file included from ../../src/third_party_impl.cc:5:
../../third_party/loguru\loguru.hpp:2311:3: error: use of undeclared identifier 'localtime_r'; did you mean 'localtime_s'?
                localtime_r(&sec_since_epoch, &time_info);
                ^
D:\msys64\mingw64\x86_64-w64-mingw32\include\time.h:261:31: note: 'localtime_s' declared here
__forceinline errno_t __cdecl localtime_s(struct tm *_Tm,const time_t *_Time) { return _localtime64_s(_Tm,_Time); }
                              ^
In file included from ../../src/third_party_impl.cc:5:
../../third_party/loguru\loguru.hpp:2311:15: error: cannot initialize a parameter of type 'struct tm *' with an rvalue of type 'time_t *' (aka 'long long *')
                localtime_r(&sec_since_epoch, &time_info);
                            ^~~~~~~~~~~~~~~~
D:\msys64\mingw64\x86_64-w64-mingw32\include\time.h:261:54: note: passing argument to parameter '_Tm' here
__forceinline errno_t __cdecl localtime_s(struct tm *_Tm,const time_t *_Time) { return _localtime64_s(_Tm,_Time); }
                                                     ^
In file included from ../../src/third_party_impl.cc:5:
../../third_party/loguru\loguru.hpp:2805:4: error: "No signal handlers on Win32" [-Werror,-W#warnings]
                #warning "No signal handlers on Win32"
                 ^
8 errors generated.

[27/71] Compiling src/clang_translation_unit.cc
Waf: Leaving directory `/d/shaman/cquery/build/release'
Build failed
 -> task in 'bin/cquery' failed with exit status 1 (run with -v to display more information)

--- a/loguru.hpp
+++ b/loguru.hpp
@@ -1308,7 +1308,7 @@ This will define all the Loguru functions so that the linker may find them.
 #include <thread>
 #include <vector>
 
-#ifdef _MSC_VER
+#ifdef _WIN32
 	#include <direct.h>
 
 	#define localtime_r(a, b) localtime_s(b, a) // No localtime_r with MSVC, but arguments are swapped for localtime_s
@@ -1625,7 +1625,7 @@ namespace loguru
 	LOGURU_PRINTF_LIKE(1, 0)
 	static Text vtextprintf(const char* format, va_list vlist)
 	{
-#ifdef _MSC_VER
+#ifdef _WIN32
 		int bytes_needed = _vscprintf(format, vlist);
 		CHECK_F(bytes_needed >= 0, "Bad string format: '%s'", format);
 		char* buff = (char*)malloc(bytes_needed+1);
@@ -1940,7 +1940,7 @@ namespace loguru
 		for (char* p = strchr(file_path + 1, '/'); p; p = strchr(p + 1, '/')) {
 			*p = '\0';
 
-	#ifdef _MSC_VER
+	#ifdef _WIN32
 			if (_mkdir(file_path) == -1) {
 	#else
 			if (mkdir(file_path, 0755) == -1) {

